cmd/snap-update-ns/utils.go

```diff
@@ -45,6 +47,99 @@ var (
 	sysFchown  = syscall.Fchown
 )
 
+// ReadOnlyFsError is an error encapsulating encountered EROFS.
+type ReadOnlyFsError struct {
+	Path string
+}
+
+func (e *ReadOnlyFsError) Error() string {
+	return fmt.Sprintf("cannot operate on read-only filesystem at %s", e.Path)
+}
+
+// Create directories for all but the last segments and return the file
+// descriptor to the leaf directory. This function is a base for secure
+// variants of mkdir, touch and symlink.
+func secureMkPrefix(segments []string, perm os.FileMode, uid, gid int) (int, error) {
+	logger.Debugf("secure-mk-prefix %q %v %d %d -> ...", segments, perm, uid, gid)
+
+	// Declare var and don't assign-declare below to ensure we don't swallow
+	// any errors by mistake.
+	var err error
+	var fd int
```

Comment:
I'd initialize it to a sane and safe value (which is -1 in this code), just to avoid nasty bugs in the future.

Reply:
This is a comment for the base PR. I'll address it there.

Reply:
I did and later undid this. It is an ineffectual assignment as all code paths later overwrite the value before it has a chance of being used. This is an error according to our validation tools.